blueprint/core/blueprint_CanvasView.h

```diff
@@ -11,9 +11,321 @@
+                        const float endAngle      = args.arguments[4];
+                        //TODO; Handle antiClockWise
+                        //bool        antiClockWise = args.arguments[5];
+
+                        const float width  = radius * 2;
+                        const float height = radius * 2;
+
+                        path.addArc(x, y, width, height, startAngle, endAngle);
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("quadraticCurveTo", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 4);
+
+                        const float controlPointX = args.arguments[0];
+                        const float controlPointY = args.arguments[1];
+                        const float endPointX     = args.arguments[2];
+                        const float endPointY     = args.arguments[3];
+
+                        path.quadraticTo(controlPointX, controlPointY, endPointX, endPointY);
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("closePath", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 0);
+
+                        path.closeSubPath();
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("stroke", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 0);
+
+                        graphics->setColour(properties.strokeStyle.colour);
+                        //TODO: Consider how to map HTML Canvas API properties to juce::PathStrokeType
+                        graphics->strokePath(path, juce::PathStrokeType((float)properties.lineWidth));
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("fill", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 0);
+
+                        graphics->setColour(properties.strokeStyle.colour);
+                        graphics->strokePath(path, juce::PathStrokeType((float)properties.lineWidth));
+                        graphics->setColour(properties.fillStyle.colour);
+                        graphics->fillPath(path);
+
+                        return juce::var();
+                    }
+            });
+
+            // Transforms
+            setProperty("rotate", juce::var::NativeFunction {
```

Comment:
Simple transform and rotation support allowed drawing of below slider like thing.

![image](https://user-images.githubusercontent.com/12968937/82732499-84679680-9d05-11ea-9acf-e0639981a7b0.png)

```
function drawCircle(ctx, x, y, radius, filled) {
  ctx.beginPath();
  ctx.moveTo(x + radius, y);
  ctx.arc(x, y, radius, 0, 2 * Math.PI);

  if (filled)
    ctx.fill();
  else
    ctx.stroke();
}

function drawRotaryKnob(ctx, x, y, radius) {

  // Draw slider thumb
  drawCircle(ctx, x, y, radius, true);

  ctx.moveTo(x + radius, y);
  ctx.lineTo(x + radius, y + (radius / 1.5));
  ctx.stroke();

  // Draw slider track
  const pad     = 20;
  const arcSize = 0.7;

  ctx.beginPath();
  ctx.moveTo((x - pad) + (radius + pad), y - pad);
  ctx.arc(x - pad, y - pad, (radius + pad), 0,  2 * Math.PI * arcSize);

  const rotateAngle = (180 + (180 * (1 - arcSize)))  * Math.PI / 180;

  const deltaX = x + radius;
  const deltaY = y + radius;

  ctx.translate(deltaX, deltaY);
  ctx.rotate(rotateAngle);
  ctx.translate(-deltaX, -deltaY);

  ctx.stroke();
}

function doCanvasDraw(ctx) {
  ctx.strokeColour = `ff884848`;
  ctx.fillColour   = `ffffffaa`;
  drawRotaryKnob(ctx, 400, 30, 40);
}
```

Reply:
Got a resetTransform() function working with the above approach. Still a couple kinks to workout.

Reply:
@nick-thompson, yeah a stack is nice idea. I effectively have this in a less elegant shape at the moment. Hitting a couple of issues with it around invert() not applying to some transforms etc. This sounds like the right track to head down though. I'll take a stab at this now and let you know how it goes.

Reply:
@nick-thompson, resolved in https://github.com/nick-thompson/blueprint/pull/49/commits/0dfb056c504738c28a7b553ae979174e47ee52f8. Tested and works great! Thanks for the tip.

Reply:
@nick-thompson, my only slight reservation with this approach is it might mean a bit of acrobatics to support getTransform(): https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getTransform

But maybe that isn't something we need to worry about right at this second and we can deal with in a future change.

Reply:
Ahh interesting. Yea, I don't think we need to support `getTransform` right now, that can be a "deal with it later" kind of thing. If our implementation supports _most_ of the Canvas Web API then I think we're in a reallly good spot.

That said, thinking about it, `getTransform`, as far as I understand it, returns a single transform that reflects the current state of the canvas context. So if you apply 5 rotations and 3 translations, they're all represented in a single affine transform matrix that is returned here. To do that we just need matrix mults of the underlying matrix coefficients. So, instead of storing the matrix inverses in our `transformStack`, we could store them as is so that we can do this:

```cpp
setProperty("rotate", juce::var::NativeFunction {
  /// etc etc
  transformStack.push_back(transform);
});

setProperty("getTransform", juce::var::NativeFunction {
    // etc etc
    Matrix<3, 3> a = Matrix<3, 3>::Identity();
    for (auto& t : transformStack) {
        a = a * makeMatrixFromAffineTransform(t);
    }
    return makeVarFromMatrix(a);
}

setProperty("resetTransform", juce::var::NativeFunction {
  /// etc etc
  while (transformStack.size() > 0) {
     g.addTransform(transformStack.pop_back().inverted();
  }
  return juce::var();
});
```

I got a little hand-wavy around the matrix multiplication, but hopefully you see what I mean. Let me know what you think! And yea I'm totally happy to defer this to a later date. If you think this sounds like a good approach we can write a new issue to track implementing the `getTransform` stuff

Reply:
Ha yeah that was the exact "acrobatics" I had in mind. It's easily sorted when we come to it.

Reply:
Yea, acrobatics for a later time 😆 

Ok great, I'll document this with a new issue